File: app/src/main/java/uk/co/mezpahlan/oldtimerag/theguardian/feed/FeedMvp.java
package uk.co.mezpahlan.oldtimerag.theguardian.feed;

import android.support.annotation.Nullable;

import java.util.List;

import uk.co.mezpahlan.oldtimerag.base.BaseMvp;
import uk.co.mezpahlan.oldtimerag.data.model.search.Result;
import uk.co.mezpahlan.oldtimerag.data.model.search.Search;

/**
 * MVP definitions for TheGuardian.Feed
 */
public interface FeedMvp {
    interface View extends BaseMvp.LCEView <List<Result>> {
        void showGuardianArticle(String id, String title);
    }

    interface Presenter extends BaseMvp.Presenter <List<Result>> {
        void load(String queryType);
        void onConfigurationChanged(View view, @Nullable String queryType);
        void onSelectResult(Result result);
    }

    interface ModelInteractor extends BaseMvp.ModelInteractor <Search> {
        void fetch(@Nullable String queryType);
        void fetchCached(@Nullable String queryType);
    }
}

File: app/src/main/java/uk/co/mezpahlan/oldtimerag/theguardian/feed/FeedMvp.kt
package uk.co.mezpahlan.oldtimerag.theguardian.feed

import uk.co.mezpahlan.oldtimerag.base.BaseMvp
import uk.co.mezpahlan.oldtimerag.data.model.search.Result
import uk.co.mezpahlan.oldtimerag.data.model.search.Search

/**
 * MVP definitions for TheGuardian.Feed
 */
interface FeedMvp {
    interface View : BaseMvp.LCEView<List<Result>> {
        fun showGuardianArticle(id: String, title: String)
    }

    interface Presenter : BaseMvp.Presenter<List<Result>> {
        fun load(queryType: String)
        fun onConfigurationChanged(view: View, queryType: String?)
        fun onSelectResult(result: Result)
    }

    interface ModelInteractor : BaseMvp.ModelInteractor<Search> {
        fun fetch(queryType: String?)
        fun fetchCached(queryType: String?)
    }
}